### DISCUSSION TITLE
Is there an easy solution for persisting input values?

### DISCUSSION THREAD

**Discussion by gouldingken** [observablehq/framework#887] [Q&A] [answered]
Currently if I switch to a different "page" within the framework, all my inputs are reset to their default values. Is there a way to persist those values while the session is live - and possibly across sessions (e.g. using localStorage or a database connection)?

**Comment by Fil** [comment, marked as answer]
Yes I think you could use _localStorage_  to save state (or _sessionStorage_ if you want two browser windows to have their own separate history, which I tend to prefer); then set the initial value of fresh Inputs to the stored value, if any.

It could go like this (in two different js code blocks):

````md
```js
const defaultColor = "#ff0000";
const color = view(Inputs.color({value: sessionStorage.getItem("color") ?? defaultColor}));
```

```js
{
  sessionStorage.setItem("color", color);
}
```
````

**Reply by mbostock** [reply]
I think we should add a “queryKey” option or some such that automatically does two-way synchronization with the URL. (Maybe we could do a “sessionKey” option too.)
